src/logger/lwt_log_core.ml

```diff
@@ -47,6 +47,17 @@ let string_of_level = function
   | Error -> "error"
   | Fatal -> "fatal"
 
+let level_of_string str =
+  let str = (String.lowercase [@ocaml.warning "-3"]) str in
+  match str with
+  | "debug" -> Ok Debug
+  | "info" -> Ok Info
+  | "notice" -> Ok Notice
+  | "warning" -> Ok Warning
+  | "error" -> Ok Error
+  | "fatal" -> Ok Fatal
+  | _ -> Pervasives.Error (Printf.sprintf "invalid log level (%s)" str)
```

Comment:
So, right now, shall we use option or introduce a dependency on the result compatibility library? If 4.02 support is a goal, option is the easiest way out, and I agree Error doesn't offer much more than None here, but I'll do whichever you tell me to do.